### PR TITLE
Add Safe Division Function

### LINKED ISSUE #41: safe division function
[`R.divide`][1] isn't actually unsafe, but `Infinity` is as unhelpful as `NaN` in many situations.

```haskell
div :: Number -> Number -> Maybe Number
div a 0 = Nothing
div a b = Just (a / b)
```

[1]: http://ramdajs.com/docs/#divide

### PR DESCRIPTION
**Pull request by svozza**
Closes #41

### CODE REVIEW

**Comment by davidchambers** [comment]
Shall we name the function `divide` to match Ramda, or `div` to match Haskell?

**Comment by svozza** [comment]
Yeah, I wasn't sure about that. Initially I was going to do something point free with `R.ifElse`and `R.divide`so that's why I named it `divide`. However, the point free version was  was complete overkill so I just went with the simple opt ion but didn't revisit the name. I'm thinking we may as well stick with the Ramda name though, it'll probably be less surprising to users.

**Comment by davidchambers** [comment]
`S.divide` sounds good to me.

File: test/index.js [hunk 1]
```diff
@@ -1033,6 +1033,18 @@ describe('control', function() {
...
 
+describe('number', function() {
+
+  it('returns a Just when the divisor is not 0', function() {
+    eq(S.divide(20, 5), S.Just(4));
+  });
+
+  it('returns Nothing when the divisor is 0', function() {
+    eq(S.divide(20, 0), S.Nothing());
+  });
```
Comment (davidchambers) [inline test/index.js]: Let's include a test case for division by the *other* zero. ;)

File: test/index.js [hunk 2]
```diff
@@ -1033,6 +1033,18 @@ describe('control', function() {
 
 });
 
+describe('number', function() {
+
+  it('returns a Just when the divisor is not 0', function() {
+    eq(S.divide(20, 5), S.Just(4));
+  });
```
Comment (davidchambers) [inline test/index.js]: It's probably worth including a test case which demonstrates that the function does not do integer division.

Comment (svozza) [inline test/index.js]: Cool. I've added some examples to the docs to reflect this as well.

File: test/index.js (hunk 1 above)
Comment (svozza) [inline test/index.js]: Haha. Done.

**Comment by davidchambers** [comment]
This looks good. I'd like to leave this pull request open, though, until I spend some time working on #45. That way we can consider all the mathematical operators in the context of both Number and Int before committing to anything.

**Comment by svozza** [comment]
Yep, sounds like a good plan.

**Comment by davidchambers** [comment]
Closing for the reason given in https://github.com/plaid/sanctuary/issues/41#issuecomment-165291887.
